iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -784,16 +784,17 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
         {
             try
             {
-                if (!_openCalled)
+                if (!_openCalled
+                    || GetClientTransportStatus() == ClientTransportStatus.Closed)
                 {
                     return;
                 }
 
                 if (Logging.IsEnabled)
                     Logging.Enter(this, cancellationToken, nameof(CloseAsync));
 
-                _handleDisconnectCts.Cancel();
-                _cancelPendingOperationsCts.Cancel();
+                _handleDisconnectCts?.Cancel();
```

Comment:
I don't see how these get to null, but we're getting an NRE on this line when closing twice. I'd rather the client get an ObjectDisposedException than NRE, but we should be protected from even that by the change on 788.